d3a_api_client/aggregator.py

```diff
@@ -14,23 +14,27 @@ def __init__(self, rest_client):
         super().__init__(rest_client)
 
     def received_message(self, message):
-        if "event" in message:
-            if message["event"] == "market":
-                self.client._on_market_cycle(message)
-            elif message["event"] == "tick":
-                self.client._on_tick(message)
-            elif message["event"] == "trade":
-                self.client._on_trade(message)
-            elif message["event"] == "finish":
-                self.client._on_finish(message)
-            elif message["event"] == "selected_by_device":
-                self.client._selected_by_device(message)
-            elif message["event"] == "unselected_by_device":
-                self.client._unselected_by_device(message)
-            else:
-                logging.error(f"Received message with unknown event type: {message}")
-        elif "command" in message:
-            self.command_response_buffer.append(message)
+        try:
+            if "event" in message:
+                if message["event"] == "market":
+                    self.client._on_market_cycle(message)
+                elif message["event"] == "tick":
+                    self.client._on_tick(message)
+                elif message["event"] == "trade":
+                    self.client._on_trade(message)
+                elif message["event"] == "finish":
+                    self.client._on_finish(message)
+                elif message["event"] == "selected_by_device":
+                    self.client._selected_by_device(message)
+                elif message["event"] == "unselected_by_device":
+                    self.client._unselected_by_device(message)
+                else:
+                    logging.error(f"Received message with unknown event type: {message}")
+            elif "command" in message:
+                self.command_response_buffer.append(message)
```

Comment:
```python
            elif "command" in message:
                self.command_response_buffer.append(message)
        if "event" in message or "command" in message:
            self.client._on_event_or_response(message)
                
```
Do you mind , as a side fix, adding the on_event_or_response callback here, I noticed that due to inheritance this one two needs to be explicitly added

Reply:
Yes, I vote for that too, thanks!

Reply:
I also promised Sarah that :-)

Reply:
Thanks for also catching that ! I'll add it now